cmd/zed/zed.d/io-spare.sh

```diff
@@ -209,22 +177,21 @@ main()
         #
         # Round-robin through the spares trying those that are available.
         #
-        for spare in ${ZEVENT_VDEV_SPARE_PATHS}; do
-
-            # shellcheck disable=SC2046
-            set -- $(query_vdev_status "${ZEVENT_POOL}" "${spare}")
-            spare_path="$1"
-            spare_status="$2"
+        for spare in $(get_pool_spares "${ZEVENT_POOL}"); do
+	    # Get the size of the spare and the failed VDEV.
+	    spare_size="$(get_dev_size "${spare}")"
+	    faild_size="$(get_krn_size "${ZEVENT_VDEV_PATH}")"
 
-            [ "${spare_status}" = "AVAIL" ] || continue
+	    # If the spare is to small, try another.
+	    [ "${spare_size}" < "${faild_size}" ] && continue
```

Comment:
This is kind'a the "magic" and the part that probably won't work as-is. Not without smarter <code>get_pool_spares()</code> and <code>get_krn_size()</code> of course.